### PR TITLE
feat: carousel component

### PR DESCRIPTION
**Pull request by vzdiogo**
# Description

this updade addresses the new carousel component. the component accepts images, video, text and custom components.
has infinite loop,  auto-play, and thumbnails.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,5 +103,8 @@
     "extends": [
       "plugin:storybook/recommended"
     ]
+  },
+  "dependencies": {
+    "react-responsive-carousel": "^3.2.23"
```
Comment (joaoprcosta) [inline package.json:108]: react-responsive-carousel must be a dependency for the package and not for the entire monorepo

File: packages/components/carousel/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# @douro-ui/button
```
Comment (joaoprcosta) [inline packages/components/carousel/README.md:1]: This is not button component

File: packages/components/carousel/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@douro-ui/carousel",
+  "version": "0.1.3",
```
Comment (joaoprcosta) [inline packages/components/carousel/package.json:3]: Must start in version 0.0.0

File: packages/components/carousel/src/__tests__/Carousel.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import Carousel from '../carousel';
+
+const defaultProps = {
+  images: [
+    {
+      src: 'data:image/svg+xml;charset=UTF-8,<svg xmlns="http://www.w3.org/2000/svg" width="300" height="300"><rect width="100%" height="100%" fill="white" stroke="black" stroke-width="2"/></svg>',
```
Comment (joaoprcosta) [inline packages/components/carousel/src/__tests__/Carousel.spec.tsx:7]: A proper way to use icons could be adding them to the icons package in Douro-ui, put that package as a dependency of this one, and release them.

File: packages/components/carousel/src/__tests__/Carousel.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    expect(screen.queryByLabelText('Next')).not.toBeInTheDocument();
+  });
+
+  it('calls onClick handler when a slide is clicked', () => {
+    const handleClick = jest.fn();
+
+    render(<Carousel {...defaultProps} onClick={handleClick} />);
+
+    fireEvent.click(screen.getByAltText('Slide 1'));
+    expect(handleClick).toHaveBeenCalled();
```
Comment (joaoprcosta) [inline packages/components/carousel/src/__tests__/Carousel.spec.tsx:50]: Also you need to expect to see the Slide 2 to complete the behaviour

File: packages/components/carousel/src/__tests__/Carousel.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  it('should be in autoplay mode when autoPlay is true', () => {
+    jest.useFakeTimers();
+    render(<Carousel {...defaultProps} autoPlay={true} interval={1000} />);
+    jest.advanceTimersByTime(1000);
```
Comment (joaoprcosta) [inline packages/components/carousel/src/__tests__/Carousel.spec.tsx:56]: Missing expect of something.

File: packages/components/carousel/src/carousel.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+          showThumbs={showThumbs}
+          infiniteLoop={infiniteLoop}
+          autoPlay={autoPlay}
+          interval={interval}
+          centerMode={true}
+          centerSlidePercentage={45}
+          {...props}
+        >
+          {images.map((image, index) => (
+            <div className="carousel-image-container" key={index}>
```
Comment (joaoprcosta) [inline packages/components/carousel/src/carousel.tsx:35]: import { ContainerStyled } from './carousel.styles'

```suggestion
            <ContainerStyled key={index}>
```

Comment (joaoprcosta) [reply packages/components/carousel/src/carousel.tsx:35]: Also add a proper index to better identify the element: 

```suggestion
            <div className="carousel-image-container" key={`carousel-container-${index}`}>
```

File: packages/components/carousel/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+export { default as DemoCarousel } from './carousel';
```
Comment (joaoprcosta) [inline packages/components/carousel/src/index.ts:1]:
```suggestion
export { default as Carousel } from './carousel';
```

File: packages/components/carousel/vite.config.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        plugins: ['@emotion/babel-plugin'],
+      },
+    }),
+  ],
+  ...viteBuildBaseConfig<BuildOptions>(
+    resolve(__dirname, mainEntry),
+    packageName,
+    {
+      outDir: '../../../dist',
+      minify: false,
```
Comment (joaoprcosta) [inline packages/components/carousel/vite.config.ts:22]: Remove minify property.

**Review by JuniorDiasOliveira — APPROVED** [review]

File: packages/svg-icons/src/index.ts [hunk 10]
```diff
@@ -1 +1,30 @@
 export const test = 'test';
+
+export const carouselContents = [
```
Comment (joaoprcosta) [inline packages/svg-icons/src/index.ts:3]: These icons are not ok. Everything inside `svg-icons` should be exported separately. This is mocked data and should be placed outside the packages folder.

File: packages/components/carousel/src/Carousel.tsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
+import React, { Component, MouseEventHandler } from 'react';
+import { Carousel as ResponsiveCarousel } from 'react-responsive-carousel';
+import 'react-responsive-carousel/lib/styles/carousel.min.css';
+import type { CarouselProps, Content } from './carousel.types';
+import { CarouselStyled, CarouselHtmlStyled } from './carousel.styles';
+
+class Carousel extends Component<CarouselProps> {
```
Comment (joaoprcosta) [inline packages/components/carousel/src/Carousel.tsx:7]: Why do you create a class component?

File: yarn.lock [hunk 12]
```diff
@@ -4460,6 +4460,11 @@ cjs-module-lexer@^1.0.0:
   resolved "https://registry.yarnpkg.com/cjs-module-lexer/-/cjs-module-lexer-1.3.1.tgz#c485341ae8fd999ca4ee5af2d7a1c9ae01e0099c"
   integrity sha512-a3KdPAANPbNE4ZUv9h6LckSl9zLsYOP4MBmhIPkRaeyybt+r4UghLvq+xw/YwUcC1gqylCkL4rdVs3Lwupjm4Q==
 
+classnames@^2.2.5:
```
Comment (joaoprcosta) [inline yarn.lock:4463]: Run again yarn install again. Some dependencies don't belong to this package.

File: packages/components/carousel/src/Carousel.tsx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+  renderContent(content: Content): React.ReactNode {
+    if ('src' in content && 'alt' in content) {
```
Comment (joaoprcosta) [inline packages/components/carousel/src/Carousel.tsx:9]: You don't need to specify what kind of content you need to receive. Each item carousel must have only children!!

File: eslint.config.mjs [hunk 14]
```diff
@@ -48,6 +50,27 @@ export default [
       ],
     },
   },
+  {
```
Comment (joaoprcosta) [inline eslint.config.mjs:53]: Why are these changes here?

File: packages/components/carousel/src/__tests__/Carousel.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+import Carousel from '../Carousel';
+
+const mockContents = [
+  { src: 'image1.jpg', alt: 'Image 1' },
+  { src: 'image2.jpg', alt: 'Image 2' },
+  { src: 'video1.mp4', type: 'video/mp4' },
+  { html: '<div>HTML Content</div>' },
+];
+
+describe('Carousel Component', () => {
```
Comment (joaoprcosta) [inline packages/components/carousel/src/__tests__/Carousel.spec.tsx:11]: The tests are useless. We don't need to test the type of content of each item. The main test must be clicking on an arrow and check if the position selected is correct!

File: packages/components/carousel/src/Carousel.tsx [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+            isSelected: boolean,
+            index: number,
+          ) => (
+            <li key={index} className="indicator-item">
+              <button
+                type="button"
+                aria-pressed={isSelected}
+                aria-label={`Slide ${index + 1}`}
+                onClick={clickHandler}
+                className={`indicator ${isSelected ? 'selected' : ''}`}
```
Comment (joaoprcosta) [inline packages/components/carousel/src/Carousel.tsx:65]: You don't need to use className.

File: packages/components/carousel/src/Carousel.tsx [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+          centerMode={true}
+          centerSlidePercentage={35}
+          {...props}
+          aria-label="Image and video carousel"
+          renderIndicator={(
+            clickHandler: MouseEventHandler<HTMLButtonElement>,
+            isSelected: boolean,
+            index: number,
+          ) => (
+            <li key={index} className="indicator-item">
```
Comment (joaoprcosta) [inline packages/components/carousel/src/Carousel.tsx:59]: You don't need to use className. Just see examples from another component.

**Review by Arthurmqz — APPROVED** [review]
